include/oxen/quic/ip.hpp

```diff
@@ -93,17 +95,19 @@ namespace oxen::quic
 
         constexpr ipv6() = default;
 
+        ipv6(const std::string& str);
+
         constexpr std::optional<ipv6> next_ip() const
         {
             // If lo will not overflow, increment and return
             if (not increment_will_overflow(lo))
-                return ipv6{{hi, lo + 1}};
+                return ipv6{std::make_pair(hi, lo + 1)};
```

Comment:
I had a thought upon seeing this: we could drop the std::pair constructor if we changed this to:
```
            if (not increment_will_overflow(lo)) {
                ipv6 next{*this};
                next.lo++;
                return next;
            }
```
and similarly for the other one that changes `hi`.

(Purely a suggested alternative: feel free to ignore if you prefer the pair approach).

Reply:
i am with you on nuking the pair ctor. Let me change its usage in the test code